### PR TITLE
fix(amazon-cognito-identity-js): specify the correct userAgent/deviceName when remembering devices (React Native)

### PR DESCRIPTION
**Pull request by Samaritan1011001**
#### Description of changes

API: `Auth.rememberDevice` & `Auth.listDevices`
Existing code only depended on the `navigator !== undefined` to check if it's the target device is a browser or not. 
React Native [initializes](https://github.com/facebook/react-native/blob/main/Libraries/Core/setUpNavigator.js#L15-L18) this `navigator` object leaving the above check invalid and inaccurate. 
Here we make use of [polyfilled](https://github.com/facebook/react-native/blob/main/Libraries/Core/setUpNavigator.js#L21) `navigator.product` to determine if it's ReactNative and update the device name accordingly

#### Description of how you validated changes
Ran a sample React Native **Android** app containing the remember and list devices feature turned on. List Devices API returned the device with the name 'react-native'

Integ test for this fix running on iOS only - https://github.com/aws-amplify/amplify-js-samples-staging/pull/491
CircleCI tests passing on a fix branch - https://app.circleci.com/pipelines/github/aws-amplify/amplify-js/17151/workflows/0997c593-9fe9-4e58-b157-9efe91110e7d

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/src/CognitoUser.js [hunk 1]
```diff
@@ -55,8 +55,11 @@ import StorageHelper from './StorageHelper';
  * @param {bool=} userConfirmationNecessary User must be confirmed.
  */
 
-const isBrowser = typeof navigator !== 'undefined';
-const userAgent = isBrowser ? navigator.userAgent : 'nodejs';
+const isNavigatorAvailable = typeof navigator !== 'undefined';
+var specificUserAgent = isNavigatorAvailable ? navigator.userAgent : '';
+if (isNavigatorAvailable && navigator.product === 'ReactNative')
+	specificUserAgent = 'react-native';
+const userAgent = isNavigatorAvailable ? specificUserAgent : 'nodejs';
```
Comment (elorzafe) [inline packages/amazon-cognito-identity-js/src/CognitoUser.js:62]: Why not use `Platform` instead.

Comment (elorzafe) [reply packages/amazon-cognito-identity-js/src/CognitoUser.js:62]: And we might to refactor a little bit to use native file to make it safer? what do you think?

**Review by elorzafe — CHANGES_REQUESTED** [review]

File: packages/amazon-cognito-identity-js/src/CognitoUser.js [hunk 2]
```diff
@@ -55,8 +56,11 @@ import StorageHelper from './StorageHelper';
  * @param {bool=} userConfirmationNecessary User must be confirmed.
  */
 
-const isBrowser = typeof navigator !== 'undefined';
-const userAgent = isBrowser ? navigator.userAgent : 'nodejs';
+const isNavigatorAvailable = typeof navigator !== 'undefined';
+var browserReactNativeAgent = Platform.isReactNative
+	? 'react-native'
+	: navigator.userAgent;
```
Comment (elorzafe) [inline packages/amazon-cognito-identity-js/src/CognitoUser.js:62]: This could be a bug in case navigator is `undefined`

**Review by elorzafe — CHANGES_REQUESTED** [review]
Thanks @Samaritan1011001 

Need to add a check that `navigator` is not `undefined`

File: packages/amazon-cognito-identity-js/src/CognitoUser.js (hunk 2 above)
Comment (Samaritan1011001) [reply packages/amazon-cognito-identity-js/src/CognitoUser.js]: Thanks for catching that. I corrected it.

**Review by elorzafe — DISMISSED** [review]
Looks good!

Thanks @Samaritan1011001 🎖️

**Review by jimblanc — APPROVED** [review]
Looks good!

**Review by AllanZhengYP — DISMISSED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -1065,6 +1065,13 @@ jobs:
     steps:
       - integ_test_rn_ios
 
+  integ_rn_ios_device_tracking:
```
Comment (elorzafe) [inline .circleci/config.yml:1068]: You need to add this also on the workflow section https://github.com/aws-amplify/amplify-js/blob/8a50266c97ecded674df5e6cf4004ba8e6a733fc/.circleci/config.yml#L1473 and also required for deploy on https://github.com/aws-amplify/amplify-js/blob/8a50266c97ecded674df5e6cf4004ba8e6a733fc/.circleci/config.yml#L1972

**Review by elorzafe — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]

**Review by wlee221 — APPROVED** [review]

**Pull request merged**